File: src/test/resources/generics/GenericArrays.cs
using System.Collections.Generic;

namespace generics {
    public class GenericArrays<T> {
        private IList<T>[] genListArray = (IList<T>[])new IList<T>[0];

        private IDictionary<T, string>[] genMapArray = (IDictionary<T, string>[])new IDictionary<T, string>[0];

        private IDictionary<T, T>[] genMapArray2 = (IDictionary<T, T>[])new IDictionary<T, T>[0];

        private IList<T[]>[] genArrayList = (IList<T[]>[])new IList<T[]>[0];

        private IList<T[][]>[] genArrayList2 = (IList<T[][]>[])new IList<T[][]>[0];
    }
}

File: src/test/resources/generics/GenericArrays.java
package generics;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.List;

public class GenericArrays<T> {
    private List<T>[] genListArray = (List<T>[]) new List[0];
    private Map<T, String>[] genMapArray = (Map<T, String>[]) new HashMap[0];
    private Map<T, T>[] genMapArray2 = (Map<T, T>[]) new HashMap[0];
    private List<T[]>[] genArrayList = (List<T[]>[]) new ArrayList[0];
    private List<T[][]>[] genArrayList2 = (List<T[][]>[]) new ArrayList[0];
}
